projects/dev/src/app/group/group-manage/pending-request/pending-request.component.ts

```diff
@@ -22,87 +45,141 @@ export class PendingRequestComponent implements OnInit, OnChanges {
     { field: "at", order: -1 },
```

Comment:
So you don't use `multiSortMeta` anymore, right?
Ah it is used as a var of the table. But it does not seem to be required anymore (the only "benefit" is to display the default sort in the table but that's not important). Of course, you would have to manage in the service that when there is no sort, it does not pass the parameter.

Reply:
We need to provide the `multiSortMeta` initially to tell the grid know the sort order and reflect it in the UI.

Reply:
But we don't need to display the default order, in most app it is like this. And it allows the backend to decide itself what's the best default order (which may change) for performances.
If we change it in the backend, we'll not think about updating the frontend, that would not be DRY anyway.

Mmhh the next comment makes me realize the same info is even repeated twice in the frontend code (in addition to a backend info already), that is definitely an error.

Reply:
But we don't need to display the default order, in most app it is like this. And it allows the backend to decide itself what's the best default order (which may change) for performances.
If we change it in the backend, we'll not think about updating the frontend, that would not be DRY anyway.

Mmhh the next comment makes me realize the same info is even repeated twice in the frontend code (in addition to a backend info already), that is definitely an error. (if one is changed and not the other, the display is not correct?😬)

Reply:
You didn't get me.
I don't care if the sort info needed for backend.
But for the front end, I need this info to show the sort status of the table according to the field.
![image](https://user-images.githubusercontent.com/51055029/82540898-31190b00-9b82-11ea-880b-5b0440ef652f.png)

Reply:
You answer to only a part of the concern.

By default, the frontend should consider there is no sort (and let the backend choose) and then allow the user to set one. Let's consider a website with tables ... hum let's pick wikipedia, if I go on https://en.wikipedia.org/wiki/FIFA_World_Cup, on the first table, the table shows no sort by default while the table is clearly sorted by year. Probably there are counter examples, but that looks a quite common case. Anyway, I would prefer this as it is DRYer and let the backend default to a efficient result for itself.

Reply:
got it.

Reply:
fixed.

Reply:
`currentSort` is still `GROUP_REQUESTS_API.defaultSort;` at init while it should be `[]`. The frontend does not need to provide a default sort.

Reply:
sure.
fixed.

Reply:
You haven't removed the constant from `api.ts` 😔

Reply:
removed it.